src/js/tooltip.js

```diff
@@ -0,0 +1,420 @@
+import Delegate from 'ftdomdelegate';
+import Viewport from 'o-viewport';
+
+import Target from './target';
+
+class Tooltip {
+
+	/**
+	 * Represents a tooltip.
+	 * @constructor
+	 * @param {HTMLElement} tooltipEl - The tooltip element in the DOM (Required)
+	 * @param {Object} opts - An options object for configuring the tooltip (Optional)
+	*/
+	constructor (tooltipEl, opts) {
+
+		if (!Tooltip._tooltips) {
+			Tooltip._tooltips = new Map();
```

Comment:
This `Map` only `set`s and `delete`s content, I've never seen a `Map` used like that before. What is the use-case for this?